test/e2e/framework/resource_usage_gatherer.go

```diff
@@ -192,6 +210,115 @@ func (w *resourceGatherWorker) singleProbe() {
 	w.dataSeries = append(w.dataSeries, data)
 }
 
+// getOneTimeResourceUsageOnNode queries the node's /stats/summary endpoint
+// and returns the resource usage of all containerNames for the past
+// cpuInterval.
+// The acceptable range of the interval is 2s~120s. Be warned that as the
+// interval (and #containers) increases, the size of kubelet's response
+// could be significant. E.g., the 60s interval stats for ~20 containers is
+// ~1.5MB. Don't hammer the node with frequent, heavy requests.
+//
+// cadvisor records cumulative cpu usage in nanoseconds, so we need to have two
+// stats points to compute the cpu usage over the interval. Assuming cadvisor
+// polls every second, we'd need to get N stats points for N-second interval.
+// Note that this is an approximation and may not be accurate, hence we also
+// write the actual interval used for calculation (based on the timestamps of
+// the stats points in ContainerResourceUsage.CPUInterval.
+//
+// containerNames is a function returning a collection of container names in which
+// user is interested in.
+func getOneTimeResourceUsageOnNode(
```

Comment:
confirmed GetOneTimeResourceUsageOnNode() is used in this package as https://cs.k8s.io/?q=GetOneTimeResourceUsageOnNode&i=nope&files=&repos=
So it is nice to move the function to here, could you write it on the PR message?